### ISSUE TITLE
Support filtering inside a Map

### ISSUE THREAD

**Issue by Totktonada** [tarantool/graphql.0#263]
GraphQL [does not support][1] the Map type natively: it defines an JSON Object with arbitrary string keys and values of certain type. So avro-schema map type is mapped into a Scalar in our implementation. OTOH, GraphQL allows to pass an argument to any field (including scalars, AFAIU). So we can add an argument(s) that will instruct the library to postprocess a map.

Using the [lua-jsonpath][2] library seems to be quite general solution, but it has one disadvantage: we cannot guarantee that a jsonpath expression will not change a shape of a query result. If we change the shape the result will not more fit an original avro-schema. So we need some more specific solution.

We can just list all fields that are need to be shown in a result:

```graphql
{
    my_collection {
        my_map(fields: ["foo", "bar"])
    }
}
```

[1]: https://github.com/facebook/graphql/issues/101
[2]: https://github.com/tarantool/lua-jsonpath

**Comment by Totktonada** [comment]
From the other side, aliases already breaks this property: the result is not more guaranteed to match an original avro-schema. Maybe shape of a result should be more or less up to a user. So we can consider jsonpath if it allows to filter inner fields w/o moving then on the upper level. But it seems it [don't][1].

[1]: https://stackoverflow.com/a/47635897

**Comment by Totktonada** [comment]
> From the other side, aliases already breaks this property: the result is not more guaranteed to match an original avro-schema.

This can be fixed, filed #269 for this.

**Comment by Totktonada** [comment]
I'm going to archive the repository. I'll proceed as follows:

* Mark all open pull requests with the `archived` label and close.
* Mark all open issues with `archived` and `wontfix` labels and close.
* Archive the repository.

Consider the following projects:

* https://github.com/tarantool/graphql
* https://github.com/tarantool/graphqlapi
* https://github.com/tarantool/graphqlide
